test/jdk/java/awt/print/PrinterJob/ValidatePage/ValidatePage.java

```diff
@@ -21,407 +21,287 @@
-
-      //Go down array of instruction strings
-
-      String printStr, remainingStr;
-      for( int i=0; i < instructions.length; i++ )
-       {
-         //chop up each into pieces maxSringLength long
-         remainingStr = instructions[ i ];
-         while( remainingStr.length() > 0 )
-          {
-            //if longer than max then chop off first max chars to print
-            if( remainingStr.length() >= maxStringLength )
-             {
-               //Try to chop on a word boundary
-               int posOfSpace = remainingStr.
-                  lastIndexOf( ' ', maxStringLength - 1 );
-
-               if( posOfSpace <= 0 ) posOfSpace = maxStringLength - 1;
-
-               printStr = remainingStr.substring( 0, posOfSpace + 1 );
-               remainingStr = remainingStr.substring( posOfSpace + 1 );
-             }
-            //else just print
-            else
-             {
-               printStr = remainingStr;
-               remainingStr = "";
-             }
-
-            instructionsText.append( printStr + "\n" );
-
-          }// while
-
-       }// for
-
-    }//printInstructions()
-
-   //DO NOT call this directly, go through Sysout
-   public void displayMessage( String messageIn )
-    {
-      messageText.append( messageIn + "\n" );
+    private static final String INSTRUCTIONS =
+            "This test is very flexible and requires much interaction.\n" +
+            "There are several buttons.\n\n" +
+            "Set Paper: if all fields are valid numbers it sets the Paper object.\n" +
+            "This is used to create arbitrary nonsensical paper sizes to help\n" +
+            "test validatePage.\n\n" +
+            "Default Page: sets a default page. This should always be valid.\n\n" +
+            "Page Setup: brings up the page dialog. You must OK this dialog\n" +
+            "for it to have any effect. You can use this to set different size,\n" +
+            "orientation and margins - which of course affect imageable area.\n\n" +
+            "Printer: Used to set the current printer. Useful because current\n" +
+            "printer affects the choice of paper sizes available.\n" +
+            "You must OK this dialog for it to have any effect.\n\n" +
+            "Validate Page:The most important button in the test.\n" +
+            "By setting nonsensical or valid papers with varying margins etc,\n" +
+            "this should always find the closest match within the limits of\n" +
+            "what is possible on the current printer.\n\n" +
+            "Print: to the current printer. Not vital for this test request.";
+
+    public static void main(String[] args) throws Exception {
+
+        if (PrinterJob.lookupPrintServices().length == 0) {
+            throw new RuntimeException("Printer not configured or available.");
+        }
+
+        PassFailJFrame.builder()
+                .instructions(INSTRUCTIONS)
+                .testUI(ValidatePage::new)
```

Comment:
```suggestion
                .instructions(INSTRUCTIONS)
                .testTimeOut(15)
                .testUI(ValidatePage::new)
```
The test requires much interaction, so the timeout should be increased accordingly.